### PR TITLE
Add assigned_clinicians field to PatientConsultation model

### PR DESCRIPTION
**Pull request by Ashesh3**
This pull request adds the `assigned_clinicians` field to the `PatientConsultation` model, allowing multiple clinicians to be assigned to a patient consultation.

Phases for the change
1. [Schema Changes](https://github.com/coronasafe/care/pull/1900)
2. [Migrating Data and Deprecating assigned_to](https://github.com/coronasafe/care/pull/1903)

Related Issue: https://github.com/coronasafe/care_fe/issues/6276

### CODE REVIEW

**Review by sainak — APPROVED** [review]

File: care/facility/models/patient_consultation.py [hunk 1]
```diff
@@ -168,6 +168,10 @@ class PatientConsultation(PatientBaseModel, ConsultationRelatedPermissionMixin):
         related_name="patient_assigned_to",
     )
 
+    assigned_clinicians = models.ManyToManyField(
+        User, related_name="patient_assigned_clinician"
```
Comment (sainak) [inline care/facility/models/patient_consultation.py:172]:
```suggestion
        User, related_name="assigned_patients"
```

**Review by rithviknishad — APPROVED** [review]

**Comment by Ashesh3** [comment]
@vigneshhari could you review this?

File: care/facility/models/patient_consultation.py [hunk 2]
```diff
@@ -338,3 +344,14 @@ def has_object_email_discharge_summary_permission(self, request):
 
     def has_object_generate_discharge_summary_permission(self, request):
         return self.has_object_read_permission(request)
+
+
+class ConsultationClinician(models.Model):
```
Comment (rithviknishad) [inline care/facility/models/patient_consultation.py:349]: BaseModel? @sainak

Comment (Ashesh3) [reply care/facility/models/patient_consultation.py:349]: Question would be, if we need to store the extra info that the BaseModel provides, `external_id, created_date, modified_date` and also soft delete for this table which is just storing the m2m relations. 🤔

Comment (sainak) [reply care/facility/models/patient_consultation.py:349]: Its okay without it, we can track the changes in events

**Review by sainak — APPROVED** [review]

**Comment by Ashesh3** [comment]
@vigneshhari could you review this?

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**